frontend/src/old-pages/Configure/Components.tsx

```diff
@@ -636,4 +637,50 @@ function IamPoliciesEditor({
   );
 }
 
-export { SubnetSelect, SecurityGroups, InstanceSelect, LabeledIcon, ActionsEditor, CustomAMISettings, RootVolume, IamPoliciesEditor }
+type HelpTextInputProps = {
+  name: string,
+  path: string[],
+  errorsPath: string[],
+  configKey: string,
+  description: string,
+  help: string,
+  placeholder: string,
+  type?: InputProps.Type,
+  setterFunction?: (path: string[], value: string) => void
+  validationFunction?: () => boolean,
+};
+
+function HelpTextInput({
+  name,
+  path,
+  errorsPath,
+  configKey,
+  description,
+  help,
+  placeholder,
+  type = "text",
+  setterFunction = setState,
+  validationFunction = () => true,
+}: HelpTextInputProps)
+{
+  let value = useState([...path, configKey]);
+  let error = useState([...errorsPath, configKey]);
+
+  return <FormField
+      label={name}
+      errorText={error}
+      description={description}>
+    <div style={{display: "flex", flexDirection: "row", alignItems: "center", justifyContent: "space-between"}}>
+      <div style={{flexGrow: 1}}>
+        <Input
+            placeholder={placeholder}
+            value={value}
+            type={type}
+            onChange={({detail}) => {setterFunction([...path, configKey], detail.value); validationFunction();}} />
```

Comment:
suggestion(non-blocking): for higher extensibility, you could expose an `onChange` prop to users of the component

You can think of this component as glorified `Input` field with the help tooltip, it doesn't need to implement any logic

Users of the component could decide to apply a validation function or not, for instance

Reply:
This makes perfect sense, will do